docs/features/test.md

```diff
@@ -0,0 +1,153 @@
+
+> I'll use the [`oryx` Feature](https://github.com/devcontainers/features/tree/2d89dc301ed834d74b07350c7d8567eee78f966d/test/oryx) as an example if you'd like a live code sample.
+
+The following `scenarios.json` defines a single test scenario named `install_dotnet_and_oryx`.  The scenario will install the `dotnet` and `oryx` Features in the target repo with the provided options. 
+
+
+##### test/oryx/scenarios.json
+```json
+{
+    "install_dotnet_and_oryx": {
+        "image": "ubuntu:focal",
+        "features": {
+            "dotnet": {
+                "version": "6",
+                "installUsingApt": "false"
+            },
+            "oryx": {}
+        }
+    }
+}
+```
+
+The test command will build a container with the config above, and then look for a `.sh` test file with the same name.  The test will pass if the container builds successfully and the `install_dotnet_and_oryx.sh` shell script exits will a successful exit code (0).
+
+##### test/install_dotnet_and_oryx.sh
+```
+#!/bin/bash
+
+set -e
+
+# Import test library for `check` command
+source dev-container-features-test-lib
+
+check "Oryx version" oryx --version
+check "Dotnet is not removed if it is not installed by the Oryx Feature" dotnet --version
+
+# Install platforms with oryx build tool
+check "oryx-install-dotnet-2.1" oryx prep --skip-detection --platforms-and-versions dotnet=2.1.30
+check "dotnet-2-installed-by-oryx" ls /opt/dotnet/ | grep 2.1
+
+....
+....
+
+# Replicates Oryx's behavior for universal image
+mkdir -p /opt/oryx
+echo "vso-focal" >> /opt/oryx/.imagetype
+
+mkdir -p /opt/dotnet/lts
+cp -R /usr/local/dotnet/current/dotnet /opt/dotnet/lts
+cp -R /usr/local/dotnet/current/LICENSE.txt /opt/dotnet/lts
+cp -R /usr/local/dotnet/current/ThirdPartyNotices.txt /opt/dotnet/lts
+
+....
+....
+
+# Report result
+reportResults
+```
+
+The flags `--global-scenarios-only`, `--skip-scenarios`, and `--skip-autogenerated` can be passed to run a subset of tests.
+
+### Global Scenarios
+
+The `test/_global` directory is a special directory that holds scenario tests not tied to a specific Feature. This directory is useful for scenarios that broadly tests several Features in a given repository.
+
+The `--global-scenarios-only` can be passed to only run the global scenarios.
+
+## dev-container-features-test-lib
+
+The `dev-container-features-test-lib` is convenience helper [defined in the CLI](https://github.com/devcontainers/cli/blob/1910ca41015c627b884ddd69ebc52d1e8cdd8cf0/src/spec-node/featuresCLI/utils.ts#L59).  This shell script can be sourced in any shell script test file when running the `devcontainer features test` command, **although it is not required**.
+
+#### `check <LABEL> <cmd>`
+Description: Executes `cmd` and prints success/failed depending on exit code (0 === success) of `cmd`.
+Note: Do not use quotes for cmd as there is no such executable as e.g. "python3 --version".
```

Comment:
Found that hard to understand, maybe: "Use quotes to include whitespace in the label or individual arguments for the command"

And then below:
Example: `check "python is available" python3 --version`
But not: `check "python is available" "python3 --version"`